spring-batch-infrastructure/src/main/java/org/springframework/batch/item/kafka/KafkaItemWriter.java

```diff
@@ -28,27 +29,36 @@
  * </p>
  *
  * @author Mathieu Ouellet
+ * @author Takaaki Shimbo
  * @since 4.2
  *
  */
 public class KafkaItemWriter<K, T> extends KeyValueItemWriter<K, T> {
 
 	private KafkaTemplate<K, T> kafkaTemplate;
 
+	/**
+	 * @since 4.3
+	 */
+	private String topic;
+
 	@Override
 	protected void writeKeyValue(K key, T value) {
 		if (this.delete) {
-			this.kafkaTemplate.sendDefault(key, null);
+			this.kafkaTemplate.send(topic, key, null);
 		}
 		else {
-			this.kafkaTemplate.sendDefault(key, value);
+			this.kafkaTemplate.send(topic, key, value);
 		}
 	}
 
 	@Override
 	protected void init() {
 		Assert.notNull(this.kafkaTemplate, "KafkaTemplate must not be null.");
-		Assert.notNull(this.kafkaTemplate.getDefaultTopic(), "KafkaTemplate must have the default topic set.");
+		if (StringUtils.isEmpty(topic)) {
```

Comment:
To make this be a non-breaking change (and to simplify the user's experience if they don't need this functionality), we'd want this to be an optional requirement.  Can you please update the PR to only use the topic name if it has been set (and use the default if it has not).

Reply:
If the topic name has not been set, the default topic of `KafkaTemplate` will be regard as a topic that the records will be appended to.

https://github.com/spring-projects/spring-batch/blob/03ef7e0c75c55a5ce3e573aab94739ffd3a8c112/spring-batch-infrastructure/src/main/java/org/springframework/batch/item/kafka/KafkaItemWriter.java#L60

> Can you please update the PR to only use the topic name if it has been set (and use the default if it has not).

I think that this PR meets this requirements.
If it does not meet , please tell me how to fix it.

Reply:
@mminella 
Would you please check the above?